compiler/src/iree/compiler/Preprocessing/TransformExtensions/PreprocessingExtensions.cpp

```diff
@@ -310,13 +310,36 @@ IREE::transform_dialect::MatchContractionOp::matchOperation(
 DiagnosedSilenceableFailure IREE::transform_dialect::MatchDimsEqualOp::apply(
     transform::TransformRewriter &rewriter,
     transform::TransformResults &results, transform::TransformState &state) {
-  ArrayRef<transform::Param> currentDimSizes =
+  ArrayRef<transform::Param> currentDimAttrs =
       state.getParams(getDimensionSizes());
-  ArrayAttr targetDimensionSizes = getExpectedValues();
+  ArrayAttr targetDimAttrs = getExpectedValues();
 
-  if (!llvm::equal(currentDimSizes, targetDimensionSizes)) {
+  if (currentDimAttrs.size() != targetDimAttrs.size()) {
     return emitSilenceableError()
-           << "Dimension sizes do not match expected values";
+           << "dimension sizes and expected values have different lengths";
+  }
+
+  auto extractDims = [](const auto &range) {
+    return llvm::map_to_vector(
+        range, [](Attribute attr) -> std::optional<int64_t> {
+          if (auto intAttr = dyn_cast<IntegerAttr>(attr)) {
+            return intAttr.getInt();
```

Comment:
If you don't want to deal with arbitrary values, you can also tighten the op definition in tablegen / move this to the verifier. That's probably a better way to go about this error mored.